tsdb/engine/tsm1/float.go

```diff
@@ -52,20 +54,25 @@ func NewFloatEncoder() *FloatEncoder {
 
 }
 
-func (s *FloatEncoder) Bytes() []byte {
-	return append([]byte{floatCompressedGorilla << 4}, s.buf.Bytes()...)
+func (s *FloatEncoder) Bytes() ([]byte, error) {
+	return append([]byte{floatCompressedGorilla << 4}, s.buf.Bytes()...), s.err
 }
 
 func (s *FloatEncoder) Finish() {
 	if !s.finished {
 		// write an end-of-stream record
+		s.finished = true
```

Comment:
Yes.  The `Finish` call writes a `NaN` as sentinel value using `Push`.  I need to mark the stream as finished so I could allow the `NaN` value to be written via `Push`.  See https://github.com/influxdb/influxdb/pull/4587/files#diff-4fcee93b376ee05d583de8af541239b8R71